sql/core/src/test/scala/org/apache/spark/sql/sources/DisableUnnecessaryBucketedScanSuite.scala

```diff
@@ -244,7 +244,8 @@ abstract class DisableUnnecessaryBucketedScanSuite
 
   test("SPARK-33075: not disable bucketed table scan for cached query") {
```

Comment:
I think this test is useless now, as we purposely go against it now. Shall we remove it?

Reply:
addressed